src/meson.build

```diff
@@ -107,7 +107,7 @@ meson.add_install_script(
       join_paths(get_option('prefix'), get_option('bindir'), 'fwupdagent')))
 endif
 
-if get_option('systemd')
+if get_option('systemd') and get_option('offline')
```

Comment:
I think there's a new matrix here that you need to check where offline is enabled but systemd is disabled. That shouldn't be a valid configuration.

Reply:
So if offline is true and systemd is false do a error() in the toplevel meson.build perhaps?

Reply:
Yeah that's what I was thinking

Reply:
Done.